worker/worker.py

```diff
@@ -1083,6 +1083,24 @@ def detect_compilers():
     return ret
 
 
+def detect_make():
+    try:
+        with subprocess.Popen(
+            ["make"],
```

Comment:
may be this could better check for the output of `make -v`, since just invoking make without arguments is an error (i.e. non-zero exit code).

Reply:
@vondele I agree to some extent but I was afraid that different OSes may have different make options. I wonder what @ppigazzini thinks.

Reply:
Actually `make` without arguments is dangerous since there could be a spurious makefile in the worker directory.

Reply:
These are the arguments for posix make https://pubs.opengroup.org/onlinepubs/009695299/utilities/make.html . Unfortunately `-v` is not among them. This does not mean it is not a good idea to use it. But it requires some testing to see if the option is supported on the platform the worker supports. In principle this could be done in CI.

Reply:
@vondele I now use 'make -v' and check in CI that the detection works (the current CI checks that the worker works correctly in quite a few environments, including the troublesome MacOS).